zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java

```diff
@@ -462,6 +495,33 @@ public HttpClient httpClient() {
 
     ClientOptionsBuilder options = new ClientOptionsBuilder()
       .decorator(HttpDecodingClient.newDecorator());
+
+    if (httpLogging() != null) {
+      LoggingClientBuilder loggingBuilder = new LoggingClientBuilder()
+        .requestLogLevel(LogLevel.INFO)
+        .successfulResponseLogLevel(LogLevel.INFO);
+      switch (httpLogging()) {
+        case HEADERS:
+          loggingBuilder.contentSanitizer(unused -> "");
+          break;
+        case BASIC:
+          loggingBuilder.contentSanitizer(unused -> "");
+          loggingBuilder.headersSanitizer(unused -> HttpHeaders.of());
+          break;
+        case BODY:
+        default:
+          break;
+      }
+      options.decorator(loggingBuilder.newDecorator());
+      if (httpLogging() == HttpLoggingLevel.BODY) {
+        options.decorator(RawContentLoggingClient.newDecorator());
+      }
+    }
+
+    if (tracing() != null) {
+      options.decorator(BraveClient.newDecorator(tracing(), "elasticsearch"));
+    }
+
     clientCustomizer().accept(options);
     HttpClientBuilder client = new HttpClientBuilder(clientUrl)
       .factory(clientFactory())
```

Comment:
I wonder if we can somehow bind auth to the client factory.  if so it makes the rest a little easier I think.

Reply:
thinking being that somehow we can get the health checks to apply the most basic and required config. otherwise we could expose a separate  config for health checks and in worst case pass same interceptor to both places.